### DISCUSSION TITLE
how to use foundry vtt data volume with user namespaces

### DISCUSSION THREAD

**Discussion by hemmick** [felddy/foundryvtt-docker#139] [Q&A]
I've followed the steps at [this helpful link](https://docs.docker.com/engine/security/userns-remap/) to create a user 'dockremap' who exists in my `/etc/docker/daemon.json` as-below:
```json
{
        "userns-remap": "dockremap",
        "experimental": false,
        "live-restore": true,
        "ipv6": false,
        "no-new-privileges": true
}
```
additionally I have /etc/subuid `dockremap:<dockremapsuid>:65536` and subgid `dockremap:<dockremapsgid>:65536`

Finally -- I have built my docker container with your prebuilt image command and am trying to run it with the following command:
`docker run --publish 30000:30000/tcp --volume <my_homedir>/vttdata:/data fvtt-prebuilt:0.7.9`

The intent is to run foundry vtt in a container, but to persist my data in my home directory.  However, when I run the above, I get the following error:

```console
Entrypoint | 2021-01-06 00:52:24 | [info] Setting data directory permissions.
chown: /data: Operation not permitted
chown: /data/Config: Operation not permitted
```

It seems that the docker container might be trying to chown data -- I have already confirmed that <my_homedir>/vttdata is chowned to 'dockremap:dockremap'   

Any idea what I might be missing, here?

**Comment by felddy** [comment]
I haven't played with remapping yet so I can't reproduce your setup here...

I'd take a look at setting `FOUNDRY_UID` and `FOUNDRY_GID` environment variables to match the values you are using in remap: 
https://github.com/felddy/foundryvtt-docker#optional

Optionally, you can disable the `chown` step entirely by setting the `CONTAINER_PRESERVE_OWNER` variable to `/data`.
